### PR TITLE
executor: remove joinBuilder

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

File: executor/merge_join.go [hunk 1]
```diff
@@ -127,37 +126,23 @@ func (rb *rowBlockIterator) nextBlock() ([]Row, error) {
...
-		return errors.Trace(lErr)
-	}
-	rErr := e.innerIter.reader.Close()
-	if rErr != nil {
-		return errors.Trace(rErr)
+	if err := e.baseExecutor.Close(); err != nil {
+		return errors.Trace(err)
 	}
 
+	e.resultBuffer = nil
```
Comment (lamxTyler) [inline executor/merge_join.go]: Why not set it first?

**Comment by coocood** [comment]
LGTM

**Pull request merged**